common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -549,12 +549,11 @@ def insert_or_update(self, asset_md):
         """
         metadata_to_insert = asset_md.to_storable()
         asset_idx = self.find(asset_md.asset_id)
-        if asset_idx is None:
-            # Add new metadata sorted into the list.
-            self.add(metadata_to_insert)
-        else:
-            # Replace existing metadata.
-            self[asset_idx] = metadata_to_insert
+        if asset_idx is not None:
+            # Delete existing metadata.
+            del self[asset_idx]
```

Comment:
This looks like a functional change to how the code operates.  Can you elaborate on why this change was made?

Reply:
They removed support for the `__setitem__' method (now throws an exception) in favor of just deleting and adding an element instead ([link to changelog](https://github.com/grantjenks/python-sortedcontainers/blob/e05634b0f0b85deae2e796a8acc1cad5961819eb/HISTORY.rst#id15)), so now we want to call the `add` method in both scenarios, and just have to handle deleting existing metadata when appropriate.